alf/algorithms/data_transformer.py

```diff
@@ -625,6 +625,27 @@ def forward(self, reward):
         return reward.clamp(*self._minmax)
 
 
+@alf.configurable
+class RewardShifting(RewardTransformer):
+    """Shift immediate rewards by a displacement of ``bias``.
+
+    Can be used as a reward shaping function passed to an algorithm
+    (e.g. ``ActorCriticAlgorithm``).
+    """
+
+    def __init__(self, observation_spec=(), bias=0.0):
+        """
+        Args:
+            observation_spec (nested TensorSpec): describing the observation in timestep
+            bias (float): displacement amount
+        """
+        super().__init__(observation_spec)
+        self._bias = bias
+
+    def forward(self, reward):
+        return reward + self._bias
```

Comment:
Though rarely used, what if the reward is multi-dimensional?

Reply:
Now added comments on the expected behavior in multi-dim reward case.
Also added ``FunctionalRewardTransformer`` to help user to implement their desired transformations, including the multi-dim case.